### PR TITLE
Split jsEnvs, creating jsEnvsTestKit, jsEnvsTestSuite.

### PR DESCRIPTION
**Pull request by nicolasstucki**
This will make it possible to publish jsEnvsTestKit (issue #2250)
to be able to test external jsEnvs like SeleniumJSEnv.
jsEnvsTestSuite contains the test implementations of jsEnvs
implemented in scala-js/scala-js.

### CODE REVIEW

File: ci/matrix.xml [hunk 1]
```diff
@@ -191,37 +191,41 @@
...
+        stubs/package jsEnvsTestSuite/test \
         ir/mimaReportBinaryIssues tools/mimaReportBinaryIssues \
-        jsEnvs/mimaReportBinaryIssues testAdapter/mimaReportBinaryIssues \
+        jsEnvs/mimaReportBinaryIssues jsEnvsTestKit/mimaReportBinaryIssues \
+        testAdapter/mimaReportBinaryIssues \
         stubs/mimaReportBinaryIssues cli/mimaReportBinaryIssues \
         irJS/mimaReportBinaryIssues toolsJS/mimaReportBinaryIssues &&
     sbt ++$scala ir/compile:doc tools/compile:doc jsEnvs/compile:doc \
-        testAdapter/compile:doc stubs/compile:doc
+        jsEnvsTestKit:compile:doc testAdapter/compile:doc stubs/compile:doc
```
Comment (sjrd) [inline ci/matrix.xml]: This should be `jsEnvsTestKit/compile:doc` (`/` instead of the first `:`)

File: project/Build.scala [hunk 2]
```diff
@@ -648,14 +649,39 @@ object Build {
...
       )
   ).dependsOn(tools)
 
+  lazy val jsEnvsTestKit: Project = Project(
+      id = "jsEnvsTestKit",
+      base = file("js-envs-test-kit"),
+      settings = (
+          commonSettings ++ publishSettings ++ fatalWarningsSettings
+          ) ++ Seq(
+        name := "Scala.js JS Envs Test Kit",
```
Comment (sjrd) [inline project/Build.scala]: Indentation issue.

File: project/Build.scala [hunk 3]
```diff
@@ -648,14 +649,39 @@ object Build {
...
+        libraryDependencies +=
+          "junit" % "junit" % "4.8.2",
+        previousArtifactSetting,
+        binaryIssueFilters ++= BinaryIncompatibilities.JSEnvsTestKit
+      )
+  ).dependsOn(tools, jsEnvs, jUnitRuntime)
```
Comment (sjrd) [inline project/Build.scala]: This should not depend on `jUnitRuntime` (which is a Scala.js project).

File: project/Build.scala [hunk 4]
```diff
@@ -648,14 +649,39 @@ object Build {
...
+
+  lazy val jsEnvsTestSuite: Project = Project(
+    id = "jsEnvsTestSuite",
+    base = file("js-envs-test-suite"),
+    settings = (
+        commonSettings ++ fatalWarningsSettings
+        ) ++ Seq(
+      name := "Scala.js JS Envs Test Suite",
```
Comment (sjrd) [inline project/Build.scala]: Indentation issue.

File: project/build.sbt [hunk 5]
```diff
@@ -33,6 +35,7 @@ unmanagedSourceDirectories in Compile ++= {
     root / "tools/shared/src/main/scala",
     root / "tools/jvm/src/main/scala",
     root / "js-envs/src/main/scala",
+    root / "js-envs-test-kit/src/main/scala",
```
Comment (sjrd) [inline project/build.sbt]: This shouldn't be needed (and therefore, nor the dependency on `junit` on line 20)

File: project/Build.scala [hunk 6]
```diff
@@ -648,14 +649,39 @@ object Build {
...
+  lazy val jsEnvsTestKit: Project = Project(
+      id = "jsEnvsTestKit",
+      base = file("js-envs-test-kit"),
+      settings = (
+          commonSettings ++ publishSettings ++ fatalWarningsSettings
+          ) ++ Seq(
+        name := "Scala.js JS Envs Test Kit",
+        libraryDependencies +=
+          "junit" % "junit" % "4.8.2",
+        previousArtifactSetting,
```
Comment (sjrd) [inline project/Build.scala]: As this is the first version with this package, it should be `None` for now. And we'll have to switch to `previousArtifactSetting` in the "Towards 0.6.10" commit.

**Comment by sjrd** [comment]
`RetryingComJSEnvTest.scala` should be in the test suite, not the test kit.

**Comment by sjrd** [comment]
That's all.

**Comment by gzm0** [comment]
Consider (probably in a second PR) to move the ` nonStringTest` and the `slowJSEnvTest` out of `NodeJSTest` as they seem to be relevant for other envs as well (`nonStringTest` for Rhino where we provide our own console and `slowJSEnvTest` for selenium for sure).

**Comment by sjrd** [comment]
The commit message should actually start with `Fix #2250:`, as this is essentially the fix to this issue.

**Comment by sjrd** [comment]
That's all.

**Comment by nicolasstucki** [comment]
Is it really fixing #2250? Or just preparing for it?

**Comment by sjrd** [comment]
Since it modifies the `publish.sh` script, it fixes it.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**